src/cc65/loadexpr.c

```diff
@@ -115,6 +115,7 @@ void LoadExpr (unsigned Flags, struct ExprDesc* Expr)
         ** field is completely contained in the lower byte, we will throw away
         ** the high byte anyway and may therefore load just the low byte.
         */
+        unsigned OrigForceCharFlag = Flags & CF_FORCECHAR;
```

Comment:
I added this, so tell me if there's a better way.

Reply:
I'll keep my feet still until this is resolved...

Reply:
What about setting up `F`  before adding the `CF_FORCECHAR` flag to `Flags`, for example right after Line 127?

Reply:
Do you mean keeping both `F` and `Flags` at the outer scope? I don't like `F` having such an unnecessarily large scope when only the `AdjustBitField` branch needs it. I also want to do the loads as `CF_FORCECHAR` if we're adjusting since it will save the top-byte manipulations.

I could imagine something like
```
unsigned BitFieldLoadFlags = 0;
if (AdjustBitField) BitFieldLoadFlags = CF_FORCECHAR;
...
get_static (Flags | BitFieldLoadFlags | ..., ...);
...
if (AdjustBitField) {
    unsigned F = Flags | CF_CONST | CF_FORCECHAR;
    // adjustment code using F
    // final g_and using Flags | CF_CONST
}
```

Reply:
You have to keep that kind of info in the large scope in some way, be it stored in `OrigForceCharFlag` , `F` or `BitFieldLoadFlags`.

Reply:
Something has to be in the outer scope, it's a question of what. Try the new version.